src/script/chem/molfile/molfile.js

```diff
@@ -286,9 +286,135 @@ Molfile.prototype.writeCTab3000 = function (molecule) { // eslint-disable-line m
 	this.writeCR(`M  V30 COUNTS ${noOfAtoms} ${noOfBonds} ${noOfSGroups} 0 ${isChiral}`);
 	this.writeAtomBlock3000(molecule);
 	this.writeBondBlock3000(molecule);
+	this.writeSGroupBlock3000(molecule);
 	this.writeCR('M  V30 END CTAB');
 };
 
+Molfile.prototype.writeSGroupBlock3000 = function (molecule) {
+	if (molecule.sgroups.size === 0) {
+		return;
+	}
+	this.writeCR('M  V30 BEGIN SGROUP');
+	var sgmap = {};
+	var cnt = 1;
+	var sgmapback = {};
+	var sgorder = molecule.sGroupForest.getSGroupsBFS();
```

Comment:
What is the data structure of `sgorder`? I assume it is an array, but what does it contain?

Reply:
sgforest#getSGroupBFS returns an array of s-group ids in the order of breadth-first search